### ISSUE TITLE
Inline script execution isn't working with Turbolinks 5

### ISSUE THREAD

**Issue by naveed-ahmad** [gemgento/rails_script#12]
FYI: Inline script execution isn't available if you are using Turbolinks 5, so rails_script won't work.
https://github.com/turbolinks/turbolinks/issues/9#issuecomment-184843135

I've wasted couple of hours debugging this, and finally decided to go with data attributes instead. Posting the solution here in case it could save time for others hopefully.

Step 1: Add controller name as data attribute in body

````html
<body data-controller="<%= controller_path.split(/\/|_/).map(&:capitalize).join('') %>">
````

Step 2: Listen on `turbolinks:load` event and initialize the controller

````coffeescript
$(document).on "turbolinks:load",  ->
   window.$this = new (App["#{$('body').data('controller')}"] || App.Base)();
````

**Comment by kpheasey** [comment]
Thank you for this.  I'll lock v0.x to Turblinnks '< 5.0' and release v1 with the changes you have recommended.

**Comment by kpheasey** [comment]
Turbolinks is an optional dependency of RailsScript, optional dependencies cannot be defined in the gemsepc.  I will note this in the installation instructions.

**Comment by kpheasey** [comment]
@naveed-ahmad This issue has been completely resolved with 1.0.  This release is now managed under a different repository - https://github.com/kpheasey/rails_script.

The `include_rails_script()` generates an html element with data attributes for the controller name and `RailsVars`.  Then `base.js` has a `turbolinks:load` listener that initializes the correct class and RailsVars.
